config/config.go

```diff
@@ -84,38 +108,46 @@ func (c *ConfigLoader[T]) populateConfiguration(cfg *T) error {
 	}
 
 	vpr := viper.New()
-	appConfigFile := filepath.Join(path, "app-config.yaml")
-	if _, err := os.Stat(appConfigFile); err == nil {
-		tmp, err := evaluateConfigWithEnvToTmp(appConfigFile)
-		if len(tmp) != 0 {
-			defer os.Remove(tmp)
-		}
-		if err != nil {
-			return err
-		}
-
-		vpr.SetConfigFile(tmp)
-		err = vpr.ReadInConfig()
-		if err != nil {
-			return fmt.Errorf("failed to read config file: %w", err)
-		}
-	}
 
-	envConfigFilename := fmt.Sprintf("app-config.%s.yaml", getAppEnv())
-	appEnvConfigFile := filepath.Join(path, envConfigFilename)
-	if _, err := os.Stat(appEnvConfigFile); err == nil {
-		tmp, err := evaluateConfigWithEnvToTmp(appEnvConfigFile)
-		if len(tmp) != 0 {
-			defer os.Remove(tmp)
-		}
-		if err != nil {
-			return err
-		}
-
-		vpr.SetConfigFile(tmp)
-		err = vpr.MergeInConfig()
-		if err != nil {
-			return fmt.Errorf("failed to merge env config: %w", err)
+	appConfigFile := fmt.Sprintf("%s.yaml", c.ConfigFileBaseName)
+	appConfigFiles := []string{appConfigFile}
+
+	// add the env specific config file if appEnv is set
+	appEnv := getAppEnv()
+	if len(appEnv) > 0 {
+		appEnvConfigFile := fmt.Sprintf("%s.%s.yaml", c.ConfigFileBaseName, appEnv)
+		appConfigFiles = append(appConfigFiles, appEnvConfigFile)
+	}
+
+	// add additional config files
+	appConfigFiles = append(appConfigFiles, c.AdditionalConfigFileNames...)
+
+	firstFile := true
+	// iterate the appConfig files to be used, read the first one and merge the rest
+	for _, configFile := range appConfigFiles {
+		absoluteConfigFilePath := filepath.Join(path, configFile)
+		if _, err := os.Stat(absoluteConfigFilePath); err == nil {
```

Comment:
```suggestion
		_, err := os.Stat(absoluteConfigFilePath)
		if err != nil {
		  return // fail fast
		}
		
		...
```

Reply:
I don't think `return` is what we want, but `continue` should work